packages/sdk/src/resolve/resolve-accounts.js

```diff
@@ -0,0 +1,120 @@
+import {pipe, isTransaction, Ok} from "@onflow/interaction"
+
+const isFn = (d) => typeof d === "function"
+const isNumber = (d) => typeof d === "number"
+const isString = (d) => typeof d === "string"
+
+const invariant = (fact, msg, ...rest) => {
+  if (!fact) {
+    const error = new Error(`INVARIANT ${msg}`)
+    error.stack = error.stack
+      .split("\n")
+      .filter((d) => !/at invariant/.test(d))
+      .join("\n")
+    console.error("\n\n---\n\n", error, "\n\n", ...rest, "\n\n---\n\n")
+    throw error
+  }
+}
+
+const accountCanFulfillRoles = (account) => {
+  if (account.role.proposer) {
```

Comment:
there are cases where this will need a signing function too, probably easiest to always expect a signing function